ksql-engine/src/test/java/io/confluent/ksql/planner/plan/DataSourceNodeTest.java

```diff
@@ -458,7 +455,150 @@ private DataSourceNode nodeWithMockTableSource() {
     return new DataSourceNode(
         realNodeId,
         dataSource,
-        "t",
-        materializedFactorySupplier);
+        "t"
+    );
+  }
+
+  private DataSourceNode buildNodeWithMockSource() {
```

Comment:
super nit: privates to the bottom of the file.

Also, consider changing to just populate the preexisting `node` field:

```java
private void givenMockSource() {
    when(dataSource.getSchema()).thenReturn(REAL_SCHEMA);
    when(dataSource.getKeyField()).thenReturn(KEY_FIELD);
    node = new DataSourceNode(
        PLAN_NODE_ID,
        dataSource,
        "name",
        schemaKStreamFactory
    );
  }
```